### ISSUE TITLE
Translate::getModuleTranslation() with locale returns string in wrong language if there is none in provided locale

### ISSUE THREAD

**Issue by marekjedrzejewski** [PrestaShop/PrestaShop#20495]
#### Describe the bug

Using $this->l() with locale provided uses last language instead of falling back to english/base string/empty string/throwing exception/whatever.
I added pl, de, fr to default english, and for example, when I have translation in polish, but doesn't have german and french, instead of english/whatever I get polish.

#### Expected behavior

I would like to get english/base string/empty string/exception thrown/something that would indicate that there is no such string in given language.

#### Steps to Reproduce

My module is working on certain products, so I thought I'll add them in code and wrote function that uses translations to insert needed products and categories to database. I run this function is run inside modules getContent() and dump its return for testing purposes. If some string doesn't have translation in given locale I get one from previous language.

```
    public function generateLanguageArrayForString($string, $module) : array
    {
        $langarr = [];
        $activeLangs = Language::getLanguages();
        foreach ($activeLangs as $lang) {
            $langarr[(int)$lang['id_lang']] = $this->l($string, $module, $lang['locale']);
        }
        return $langarr;
    }
```
For example, when ran on "Category name" which has only polish and english translations, I get this:
```
array(4) {
  [1]=>
  string(13) "Category name"
  [6]=>
  string(15) "Nazwa kategorii"
  [7]=>
  string(15) "Nazwa kategorii"
  [8]=>
  string(15) "Nazwa kategorii"
}
```

#### Additional information
When `global $_MODULES, $_MODULE, $_LANGADM;` inside _Translate.php_'s _getModuleTranslation()_
 is changed into `$_MODULES = array(); $_MODULE = array(); $_LANGADM = array();` I get translations only for languages that have them, otherwise I get base string (but when I add english translation that is different from base string, I don't get this one, which is pretty weird). But apart from weird issue with english, the _global_ must be there for a reason, and I don't want to change things outside of module's code for this to work.

**Comment by florine2623** [comment]
Hi @marekjedrzejewski ,
Thanks for your report.
Is it possible that you send me your module so I can see what exactly is wrong ?
Thanks!

**Comment by marekjedrzejewski** [comment]
I'm not sure I could send the original module, but I extracted the important part into empty module. When module configuration is entered, you get this:
![Zrzut ekranu z 2020-08-08 09-49-59](https://user-images.githubusercontent.com/6793160/89705317-a6fc6980-d95c-11ea-99d3-e3135c909a75.png)
6 is id of pl in my shop, 7 and 8 are de and fr.

[langtest.zip](https://github.com/PrestaShop/PrestaShop/files/5044830/langtest.zip)

**Comment by khouloudbelguith** [comment]
Hi @marekjedrzejewski,

Your ticket is a duplicate of #17125
So, I close it.
Thanks!

**Comment by khouloudbelguith** [comment]
Duplicate of #17125

**Comment by marekjedrzejewski** [comment]
Doesn't seem like a duplicate to me. The other one is about strings that are in database after adding new language. Here it's about translations using last language (not the first one) for module, when nothing with languages in shop is changed. And I'm not even touching database (I plan to, but the issue is not about this). It's about behavior of _getModuleTranslation()_ which doesn't look entirely correct to me.

**Comment by khouloudbelguith** [comment]
@marekjedrzejewski, I tried with your module, I have this
![image](https://user-images.githubusercontent.com/16067358/89800285-dd89de00-db2e-11ea-824a-4658fc150de6.png)

My languages are
![image](https://user-images.githubusercontent.com/16067358/89798520-90a50800-db2c-11ea-8d90-059651e5601f.png)

Thanks!

**Comment by marekjedrzejewski** [comment]
> @marekjedrzejewski, I tried with your module, I have this
> ![image](https://user-images.githubusercontent.com/16067358/89798455-7f5bfb80-db2c-11ea-89df-c86689748a2a.png)

Can't see this image. All alpha channel shows up for me.

**Comment by khouloudbelguith** [comment]
@marekjedrzejewski, I just updated my last comment.

Thanks!

**Comment by marekjedrzejewski** [comment]
Did you provide russian translation to it? If you added another language without translating anything would it show english or russian one?

**Comment by khouloudbelguith** [comment]
@marekjedrzejewski, it shows in Russian the default language as reported here: https://github.com/PrestaShop/PrestaShop/issues/17125
https://drive.google.com/file/d/1ua8oCgVv6hAjM7nfonh8jGMDwW0FzXjM/view?usp=sharing

Thanks!

**Comment by marekjedrzejewski** [comment]
Well, that looks more like thing I'm looking for, but doesn't work for me that way.

I added russian and finnish and 'translated' the french string and I get this:

`array ( 1 => 'Translation fallback test', 6 => 'Test rezerwowych tłumaczeń', 7 => 'Test rezerwowych tłumaczeń', 8 => 'translation french', 9 => 'translation french', 10 => 'translation french', )`

Maybe there were some changes, cause I see you have version 1.7.6.7, I checked out develop and it shows 1.7.8.0 for me.

**Comment by SimonGrn** [comment]
Hello, 

Please check your problem in 1.7.6.7 which is the latest available version. If it's fixed in 1.7.6.7 we will close this issue.

Thanks!

**Comment by marekjedrzejewski** [comment]
> Hello,
> 
> Please check your problem in 1.7.6.7 which is the latest available version. If it's fixed in 1.7.6.7 we will close this issue.
> 
> Thanks!

I just checked with this: https://github.com/PrestaShop/docker, which has the 1.7.6.7.

For languages 
1-pl, 2-fr, 3-hr, 4-en, 5-es, 6-cs, 7-sv
I get: 
`array ( 1 => 'Test rezerwowych tłumaczeń', 2 => 'Test rezerwowych tłumaczeń', 3 => 'Test rezerwowych tłumaczeń', 4 => 'Langtest', 5 => 'Langtest', 6 => 'Langtest', 7 => 'Langtest', ) `

So the same behavior as with later versions.

Btw: I added english translation to test it as in this instance I had polish set as primary language.

**Comment by marekjedrzejewski** [comment]
> When `global $_MODULES, $_MODULE, $_LANGADM;` inside _Translate.php_'s _getModuleTranslation()_
> is changed into `$_MODULES = array(); $_MODULE = array(); $_LANGADM = array();` I get translations only for languages that have them, otherwise I get base string (but when I add english translation that is different from base string, I don't get this one, which is pretty weird). But apart from weird issue with english, the _global_ must be there for a reason, and I don't want to change things outside of module's code for this to work.

I found out that if I only comment out $_MODULES from global, but leave rest as is I get translations in the form I would expect - whatever has translation shows the translation, otherwise it's showing original string. This includes English, unlike when everything was made local.

` array ( 1 => 'ONLY TRUE ENGLISH TRANSLATION', 2 => 'Translation fallback test', 3 => 'Test rezerwowych tłumaczeń', 4 => 'Translation fallback test', 5 => 'Translation fallback test', 6 => 'Translation fallback test', ) `

Still, it has to be global for a reason.

As a workaround, global $_MODULES can be cleared in each iteration of a loop inside the module, to prevent from returning wrong strings, without touching core code.

**Comment by khouloudbelguith** [comment]
@marekjedrzejewski, could you please provide us clear steps to reproduce the issue
1. which language/country did you used during the install?
2. What is your language by default?
3. What is your employee's language?
4. did you add some other language?

@marekjedrzejewski, I think your issue is similar to this: https://github.com/PrestaShop/PrestaShop/issues/13024? is it?
Thanks!

**Comment by marekjedrzejewski** [comment]
> @marekjedrzejewski, could you please provide us clear steps to reproduce the issue
> 
>     1. which language/country did you used during the install?

I used docker-compose up after cloning prestashop develop branch, which comes with english and has everything set up already.
I also tested with https://github.com/PrestaShop/docker and set up polish then, issue was still there.

>     2. What is your language by default?
>     3. What is your employee's language?

Polish.

>     4. did you add some other language?
I did, as can be seen throughout my comments. Only for testing purposes though. Nothing more done with them. 

The easiest way to test it is to docker-compose up develop branch, add bunch of languages with polish somewhere between them and go to configuration page of sample module.

> @marekjedrzejewski, I think your issue is similar to this: #13024? is it?

Looks pretty simmilar to me :)
